File: customer-webapp/src/types/message.ts
export type SenderType = 'VISITOR' | 'AGENT' | 'SYSTEM';

export interface Message {
  id: number;
  conversationId: number;
  senderType: SenderType;
  senderId?: string;
  content: string;
  clientMessageId?: string;
  sequenceNumber: number;
  createdAt: string;
}

export interface SendMessageRequest {
  content: string;
  senderType: SenderType;
  clientMessageId: string;
}

File: customer-server/src/main/java/com/opencustomer/server/entity/Message.java
package com.opencustomer.server.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.opencustomer.server.enums.SenderType;
import jakarta.validation.constraints.Size;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("messages")
public class Message {
    private Long id;
    private Long conversationId;
    private SenderType senderType;
    private String senderId;
    
    @Size(max = 5000, message = "Message content must be at most 5000 characters")
    private String content;
    
    private String clientMessageId;
    private Long sequenceNumber;
    private LocalDateTime createdAt;
}
